internal/services/market.go

```diff
@@ -1,18 +1,104 @@
 package services
 
+import (
+	"context"
+	"errors"
+
+	"github.com/gofrs/uuid"
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/shopspring/decimal"
+	"github.com/shulganew/gophermart/internal/model"
+	"go.uber.org/zap"
+)
+
 // User registration service
 type Market struct {
 	stor MarketPlaceholder
 }
 
 type MarketPlaceholder interface {
-	GetOrder()
+	GetOrders(ctx context.Context, userID *uuid.UUID) ([]model.Order, error)
+	SetOrder(ctx context.Context, userID *uuid.UUID, order string, isPreorder bool, withdraw *decimal.Decimal) error
+	IsExistForUser(ctx context.Context, userID *uuid.UUID, order string) (isExist bool, err error)
+	IsExistForOtherUsers(ctx context.Context, userID *uuid.UUID, order string) (isExist bool, err error)
+	GetAccruals(ctx context.Context, userID *uuid.UUID) (accrual *decimal.Decimal, err error)
+	GetWithdrawns(ctx context.Context, userID *uuid.UUID) (withdrawn *decimal.Decimal, err error)
+	Withdrawals(ctx context.Context, userID *uuid.UUID) ([]model.Withdrawals, error)
+	IsPreOrder(ctx context.Context, userID *uuid.UUID, order string) (isPreOrder bool, err error)
+	MovePreOrder(ctx context.Context, order *model.Order) (err error)
 }
 
 func NewMarket(stor MarketPlaceholder) *Market {
-	return &Market{}
+	return &Market{stor: stor}
+}
+
+func (m *Market) SetOrder(ctx context.Context, isPreOrder bool, order *model.Order) (existed bool, err error) {
+	// Add order to the database.
+	err = m.stor.SetOrder(ctx, order.UserID, order.Onumber, isPreOrder, order.Bonus.Used)
```

Comment:
```suggestion
	err = m.stor.AddOrder(ctx, order.UserID, order.Onumber, isPreOrder, order.Bonus.Used)
```
set не равно add. Есть существенное различие, set должен перезаписать состояние до переданного у существующего или не существующего заказа

Reply:
Запомнил, исправил.